docs/framework/wcf/feature-details/wcf-discovery-overview.md

```diff
@@ -115,39 +115,39 @@ class Client
         }
     }  
   
-    // ** DISCOVERY ** //
-    static bool FindService()
-    {
-        Console.WriteLine("\nFinding Calculator Service ..");
-        DiscoveryClient discoveryClient = 
-            new DiscoveryClient(new UdpDiscoveryEndpoint());
-
-        Collection<EndpointDiscoveryMetadata> calculatorServices = 
-            discoveryClient.Find(new FindCriteria(typeof(ICalculator)));
-
-        discoveryClient.Close();
-
-        if (calculatorServices.Count == 0)
-        {
-            Console.WriteLine("\nNo services are found.");
-            return false;
-        }
-        else
-        {
-            serviceAddress = calculatorServices[0].EndpointAddress;
-            return true;
-        }
-    }
-
-    static void InvokeService()
-    {
-        Console.WriteLine("\nInvoking Calculator Service at {0}\n", serviceAddress);
-
-        // Create a client
-        CalculatorClient client = new CalculatorClient();
-        client.Endpoint.Address = serviceAddress;
-        client.Add(10,3);
-}
+    // ** DISCOVERY ** //  
+    static bool FindService()  
+    {  
+        Console.WriteLine("\nFinding Calculator Service ..");  
+        DiscoveryClient discoveryClient =   
+            new DiscoveryClient(new UdpDiscoveryEndpoint());  
+  
+        Collection<EndpointDiscoveryMetadata> calculatorServices =   
+            (Collection<EndpointDiscoveryMetadata>)discoveryClient.Find(new FindCriteria(typeof(ICalculator)));  
```

Comment:
This cast still fails. `DiscoveryClient.Find` returns a `FindResponse` object; the code should retrieve its `FindResponse.Endpoints` property and cast that to a `Collection<EndpointDiscoveryMetadata>`.

Reply:
Absolutely correct that Endpoints is missing at the end of the expression. I have committed that change